React/my-app/src/App.js

```diff
@@ -1,19 +1,78 @@
-import React, { Component } from 'react';
-import logo from './logo.svg';
-import data from './data.json'
-import './App.css';
+import React, { Component } from "react";
+import data from "./data.json";
+import "./App.css";
+import defaultImg from "./photos/default.png";
 
 class App extends Component {
+  state = { books: data, searchType: null };
+
   render() {
     return (
-      <div className="App">
-        <div className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h2>Welcome to React</h2>
+      <div className="mainContainer">
+        <h1 className="main_title"> wBooks </h1>
+        <form>
+          <select
+            className="selector"
+            name="select"
+            onChange={this.setSearchType}
+          >
+            <option value="null">Seleccionar filtro</option>
+            <option value="title">Titulo</option>
+            <option value="author">Autor</option>
+          </select>
+          {/* <img className="arrow" src="./photos/arrow.png" /> */}
+          <input
+            className="search"
+            type="text"
+            placeholder="Buscar..."
+            name="fname"
+            onChange={this.filterBooks}
+          />
+          {/* <img className="square" src="./photos/green_square.png" /> */}
+        </form>
+        <div className="gallery">
+          {this.state.books.map(element => <Book book={element} />)}
         </div>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+      </div>
+    );
+  }
+
+  setSearchType = search => {
+    const newSearchType = search.target.value;
+    this.setState({ searchType: newSearchType });
+  };
+
+  filterBooks = filterWord => {
+    const word = filterWord.nativeEvent.target.value;
+    const type = this.state.searchType;
+    if (type != "null") {
+      this.setState({
+        books: data.filter(book =>
+          book[type].toLowerCase().includes(word.toLowerCase())
+        )
+      });
+    } else {
+      this.setState({
+        books: data.filter(
+          book =>
+            book.title.toLowerCase().includes(word.toLowerCase()) ||
+            book.author.toLowerCase().includes(word.toLowerCase())
+        )
+      });
+    }
+  };
+}
+
+class Book extends Component {
+  render() {
+    const image = this.props.book.image_url;
+    const title = this.props.book.title;
+    const author = this.props.book.author;
```

Comment:
check this https://developer.mozilla.org/es/docs/Web/JavaScript/Referencia/Operadores/Destructuring_assignment